### PR TITLE
Made a generic script to pick up all the related react projects to build

### PR DESCRIPTION
**Pull request by ejazefe123**

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -127,12 +127,39 @@ install_yarn "$BUILD_DIR/.heroku/yarn" "$yarn_engine"
...
-  yarn --cwd "$BUILD_DIR/lender_onboarding_ui" build
+# get the names of all the folders in the monorepo and store them in an array
+folders=()
+while IFS='' read -r line; do
+  folders+=("$line");
+  done < <( cd "$BUILD_DIR" && ls -d -- */)
+
+if [ -e "$BUILD_DIR/package.json" ]; then
+  # get the yarn workspaces folders/patterns and store them in an array
+  workspace_patterns=( $( jq -r ".workspaces[]" < "$BUILD_DIR/package.json" ) )
```
Comment (ainscore-st) [inline bin/compile]: Do we know if `jq` will be present?

File: bin/compile [hunk 2]
```diff
@@ -127,12 +127,39 @@ install_yarn "$BUILD_DIR/.heroku/yarn" "$yarn_engine"
...
+
+  # removing the yarn workspaces folder from the main directory folder array
+  filtered_folders="${folders[@]}"
+  for ws_folder in ${workspace_patterns[@]}
+    do
+        f=$ws_folder/
+       filtered_folders=( "${filtered_folders[@]/$f}" )
+    done
+
+  install_and_build ${filtered_folders[@]}
```
Comment (ainscore-st) [inline bin/compile]: Why would we build the filtered folders?

Comment (ejazefe123) [reply bin/compile]: These are all folders excluding workspaces folders like "st_prettier_config".  I believe we don't need to install and build dependencies of workspaces on heroku. Please correct me if i am wrong.

File: bin/compile (hunk 1 above)
Comment (ejazefe123) [reply bin/compile]: No we dont know. Thanks for pointing it. Adding code to check jq and install if not present.

Comment (ejazefe123) [reply bin/compile]: Done

**Comment by ainscore-st** [comment]
@ejazefe123 there are a few issues here preventing this from working. This needs to run in the heroku environment, where `brew` will not be present, also the assumption that we want to build any directory with a package.json except those listed in the root package.json is not necessarily valid.  I'd rather not have this hold up our progress on this project, so let's instead close this PR and create a new PR where we just duplicate the lines for `lender_onboarding_ui` for `integration_admin_ui`.

**Comment by ejazefe123** [comment]
> @ejazefe123 there are a few issues here preventing this from working. This needs to run in the heroku environment, where `brew` will not be present. Also the assumption that we want to build any directory with a package.json except those listed in the root package.json is not necessarily valid.
> 
> I'd rather not have this hold up our progress on this project, so let's instead close this PR and create a new PR where we just duplicate the lines for `lender_onboarding_ui` for `integration_admin_ui`.

Sure Andy.

**Pull request closed without merging**